### ISSUE TITLE
keras.layers.Dropout  --- train mode & test mode

### ISSUE THREAD

**Issue by hccho2** [NLP-kr/tensorflow-ml-nlp#15]
CNN_Calssification, RNN Classification에서
keras.layers.Dropout은 train mode에서는 적용해야 하지만,
test mode에서는 적용이 되지 않게 모델이 구성되어야 합니다.

before:
dropout_emb = keras.layers.Dropout(rate=0.5)(embedding_layer)

after:
dropout_emb = keras.layers.Dropout(rate=0.5)(embedding_layer,training = mode == tf.estimator.ModeKeys.TRAIN)

**Comment by changwookjun** [comment]
관심 가져 주셔서 감사합니다. 

텐서플로우 core 소스를 보면 

https://github.com/tensorflow/tensorflow/blob/r1.14/tensorflow/python/keras/layers/core.py#L111-L175
@keras_export('keras.layers.Dropout')
class Dropout(Layer):
  """Applies Dropout to the input.
  Dropout consists in randomly setting
  a fraction `rate` of input units to 0 at each update during training time,
  which helps prevent overfitting.
  Arguments:
    rate: Float between 0 and 1. Fraction of the input units to drop.
    noise_shape: 1D integer tensor representing the shape of the
      binary dropout mask that will be multiplied with the input.
      For instance, if your inputs have shape
      `(batch_size, timesteps, features)` and
      you want the dropout mask to be the same for all timesteps,
      you can use `noise_shape=(batch_size, 1, features)`.
    seed: A Python integer to use as random seed.
  Call arguments:
    inputs: Input tensor (of any rank).
    training: Python boolean indicating whether the layer should behave in
      training mode (adding dropout) or in inference mode (doing nothing).
  """
위와 같이 동작하도록 만들어진 API 입니다.

이슈로 등록 해주셔서 감사합니다.

**Comment by changwookjun** [comment]
안녕하세요 . 우선 관심가져주시고 이렇게 issue달아 주셔서 감사합니다. 
https://github.com/tensorflow/tensorflow/blob/r1.14/tensorflow/python/keras/layers/core.py#L111-L175
위 tensorflow core에 보면 아래와 같은  dropout이 존재합니다.
1. 
@keras_export('keras.layers.Dropout')
class Dropout(Layer):

2. dropout이 호출 될때 
  def call(self, inputs, training=None):
    if training is None:
      training = K.learning_phase()
위에 함수에 들어오는데요   이때 training paramater에 아무런 설정을 안해주면 
3. learning_phase 에 Global 값을 가지고 있는데요 .
The learning phase flag is a bool tensor (0 = test, 1 = train)
이 값을 전달하게 됩니다.

그래서 위처럼 해주는 것도 맞으나 아무런 값을 설정안하면 자동으로 동작합니다. 

감사합니다.

**Comment by hccho2** [comment]
제가 잘못 이해하고 있었네요.
답변 감사합니다.

**Comment by Taekyoon** [comment]
창욱님께서 좋은 답변을 해주셔서 충분한 답이 되셨을 거로 보입니다.
이 이슈는 close 하겠습니다.
